src/hooks/useDetectOutsideClick.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react'
+
+/**
+ * Hook for handling closing when clicking outside of an element
+ * @function useDetectOutsideClick
+ * @param {React.node} ref
+ * @param {function} function to call on outside click
+ * @returns {[isActive, el]} isActive state and el
+ */
+export const useDetectOutsideClick = (ref, handler) => {
+  // const [isActive, setIsActive] = useState(initialState)
```

Comment:
delete @returns from jsDoc and delete commented code, or uncomment it in case it needed